sos/upload/targets/redhat.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2024 Red Hat, Inc. Jose Castillo <jcastillo@redhat.com>
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import os
+import json
+from sos.upload.targets import UploadTarget
+from sos.utilities import convert_bytes, TIMEOUT_DEFAULT
+from sos.policies.auth import DeviceAuthorizationClass
+from sos.policies.distros.redhat import RHELPolicy
+from sos import _sos as _
+
+try:
+    import requests
+    REQUESTS_LOADED = True
+except ImportError:
+    REQUESTS_LOADED = False
+
+
+class RHELUpload(UploadTarget):
+
+    client_identifier_url = "https://sso.redhat.com/auth/"\
+        "realms/redhat-external/protocol/openid-connect/auth/device"
+    token_endpoint = "https://sso.redhat.com/auth/realms/"\
+        "redhat-external/protocol/openid-connect/token"
+    upload_target_name = 'Red Hat Upload Profile'
+
+    def __init__(self, parser=None, args=None, cmdline=None):
+
+        super().__init__(parser=parser, args=args, cmdline=cmdline)
+
+    RH_API_HOST = "https://api.access.redhat.com"
+    RH_SFTP_HOST = "sftp://sftp.access.redhat.com"
+    _upload_url = RH_SFTP_HOST
+    _upload_method = 'post'
+    _device_token = None
+    # Max size for an http single request is 1Gb
+    _max_size_request = 1073741824
+
+    def check_distribution(self):
+        """Return true if we are running in a RHEL system"""
+        return isinstance(self.commons['policy'], RHELPolicy)
+
+    def pre_work(self, hook_commons):
+        caseid = self.commons['cmdlineopts'].case_id if \
```

Comment:
I think you need to call parent class `pre_work` here as well..?

Reply:
Done